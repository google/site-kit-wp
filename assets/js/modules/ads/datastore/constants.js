/**
 * `modules/ads` data store constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const MODULES_ADS = 'modules/ads';

export const MODULE_SLUG_ADS = 'ads';

export const ADWORDS_SCOPE = 'https://www.googleapis.com/auth/adwords';
export const SUPPORT_CONTENT_SCOPE =
	'https://www.googleapis.com/auth/supportcontent';

// Date range offset days for Ads report requests.
export const DATE_RANGE_OFFSET = 1;

export const ADS_MODULE_SETUP_BANNER_PROMPT_DISMISSED_KEY =
	'ads_module_setup_banner_prompt_dismissed_key';

export const ADS_WOOCOMMERCE_REDIRECT_MODAL_CACHE_KEY = 'wc-redirect-modal';

export const PLUGINS = {
	WOOCOMMERCE: 'woocommerce',
	GOOGLE_FOR_WOOCOMMERCE: 'google-listings-and-ads',
};

export const AVAILABLE_PLUGINS = [
	PLUGINS.WOOCOMMERCE,
	PLUGINS.GOOGLE_FOR_WOOCOMMERCE,
];
