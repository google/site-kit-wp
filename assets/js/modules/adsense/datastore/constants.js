/**
 * `modules/adsense` data store constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const MODULES_ADSENSE = 'modules/adsense';

export const MODULE_SLUG_ADSENSE = 'adsense';

// Date range offset days for AdSense report requests.
export const DATE_RANGE_OFFSET = 1;

export const API_STATE_READY = 'READY';
export const API_STATE_NEEDS_ATTENTION = 'NEEDS_ATTENTION';
export const API_STATE_REQUIRES_REVIEW = 'REQUIRES_REVIEW';
export const API_STATE_GETTING_READY = 'GETTING_READY';

// State to determine whether background submission is allowable.
export const BACKGROUND_SUBMIT_SUSPENDED = 'background-submit-suspended';

export const AD_BLOCKING_FORM_SETTINGS = 'adsenseAdBlockingFormSettings';

export const AD_BLOCKING_RECOVERY_SETUP_CREATE_MESSAGE_CTA_CLICKED =
	'googlesitekit-ad-blocking-recovery-setup-create-message-cta-clicked';

export const AD_BLOCKING_RECOVERY_MAIN_NOTIFICATION_KEY =
	'ad-blocking-recovery-notification';

// Various ad blocking recovery setup statuses.
export const ENUM_AD_BLOCKING_RECOVERY_SETUP_STATUS = {
	TAG_PLACED: 'tag-placed',
	SETUP_CONFIRMED: 'setup-confirmed',
};

// Zero-based index of the available ad blocking recovery setup steps.
export const ENUM_AD_BLOCKING_RECOVERY_SETUP_STEP = {
	PLACE_TAGS: 0,
	CREATE_MESSAGE: 1,
	COMPLETE: 2,
};
