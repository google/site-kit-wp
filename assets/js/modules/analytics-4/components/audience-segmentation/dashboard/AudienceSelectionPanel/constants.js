/**
 * AudienceSelectionPanel components - constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const AUDIENCE_ADD_GROUP_NOTICE_SLUG =
	'audience-segmentation-add-group-notice';
export const AUDIENCE_SELECTION_PANEL_OPENED_KEY =
	'googlesitekit-audience-selection-panel-opened';
export const AUDIENCE_SELECTION_FORM = 'audience-selection-form';
export const AUDIENCE_SELECTED = 'audience-selected';
export const AUDIENCE_SELECTION_CHANGED = 'audience-selection-changed';
export const AUDIENCE_CREATION_NOTICE_SLUG =
	'audience-segmentation-creation-notice';
export const AUDIENCE_CREATION_SUCCESS_NOTICE_SLUG =
	'audience-segmentation-creation-success-notice';
export const MIN_SELECTED_AUDIENCES_COUNT = 1;
export const MAX_SELECTED_AUDIENCES_COUNT = 3;
export const AUDIENCE_CREATION_EDIT_SCOPE_NOTICE_SLUG =
	'audience-creation-edit-scope-notice';
export const AUDIENCE_CREATION_FORM = 'audience-creation-form';
