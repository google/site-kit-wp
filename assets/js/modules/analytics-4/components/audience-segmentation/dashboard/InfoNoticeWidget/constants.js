/**
 * Constants related to audience segmentation info notices.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * UI key to display audience segmentation info notice.
 *
 * @since 1.135.0
 * @private
 */
export const AUDIENCE_INFO_NOTICE_HIDE_UI =
	'audience-segmentation-info-notice-ui';

/**
 * Slug for audience segmentation info notice.
 *
 * @since 1.130.0
 * @private
 */
export const AUDIENCE_INFO_NOTICE_SLUG = 'audience-segmentation-info-notice';

// Notice slugs: new-visitors, compare-metrics, custom-audiences, purchasers, returning-visitors, compare-new-returning, or compare-cities

/**
 * List of info notices for audiences.
 *
 * @since 1.130.0
 * @private
 */
export const AUDIENCE_INFO_NOTICES = [
	{
		slug: 'new-visitors',
		content: __(
			'The higher the portion of new visitors you have, the more your audience is growing. Looking at what content brings them to your site may give you insights on how to reach even more people.',
			'google-site-kit'
		),
	},
	{
		slug: 'compare-metrics',
		content: __(
			'Select up to three visitor groups to display on the dashboard and easily compare metrics between them.',
			'google-site-kit'
		),
	},
	{
		slug: 'custom-audiences',
		content: __(
			'Configure your own custom audiences in Analytics to gain deeper insights into visitor behavior, for example consider creating an “Existing customers” or “Subscribers” segment, depending on what goals you have for your site.',
			'google-site-kit'
		),
	},
	{
		slug: 'purchasers',
		content: __(
			'Select the Purchasers visitor group to gain insights into which visitors bring the most revenue to your site.',
			'google-site-kit'
		),
	},
	{
		slug: 'returning-visitors',
		content: __(
			'The more returning visitors your site has, the stronger and more loyal an audience you’re building. Check which content brings people back to your site - it might help you create a strategy to build a community.',
			'google-site-kit'
		),
	},
	{
		slug: 'compare-new-returning',
		content: __(
			'Compare the ratio of “new” to “returning” visitors – this can give you insights on whether you have more people stopping by as a one-off, or more loyal visitors.',
			'google-site-kit'
		),
	},
	{
		slug: 'compare-cities',
		content: __(
			'Check the cities which bring you more new vs more returning visitors – there might be new audiences you could engage with in locations you hadn’t thought about.',
			'google-site-kit'
		),
	},
];
