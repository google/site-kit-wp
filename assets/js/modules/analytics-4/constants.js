/**
 * GA4 module constants.
 *
 * Site Kit by Google, Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const ACTIVATION_STEP_REMINDER = 0;
export const ACTIVATION_STEP_SETUP = 1;
export const ACTIVATION_STEP_SUCCESS = 2;
export const ACTIVATION_STEP_IN_PROGRESS = 3;

export const ACTIVATION_ACKNOWLEDGEMENT_TOOLTIP_STATE_KEY =
	'activation-acknowledgement-tooltip-state';

export const GA4_ACTIVATION_BANNER_STATE_KEY = 'ga4-activation-banner-state';

export const ENHANCED_MEASUREMENT_ACTIVATION_BANNER_TOOLTIP_STATE_KEY =
	'enhanced-measurement-activation-banner-tooltip-state';
export const LEGACY_ENHANCED_MEASUREMENT_ACTIVATION_BANNER_DISMISSED_ITEM_KEY =
	'enhanced-measurement-activation-banner-dismissed-item';

export const REPORT_ARGS_SELECTED_METRIC_KEY = '_r.explorerCard..selmet';
export const REPORT_ARGS_SELECTED_DIMENSION_KEY = '_r.explorerCard..seldim';
export const REPORT_ARGS_DATA_FILTERS_KEY = '_r..dataFilters';
export const REPORT_ARGS_NAV_KEY = '_r..nav';

export const KM_CONNECT_GA4_CTA_WIDGET_DISMISSED_ITEM_KEY =
	'key-metrics-connect-ga4-cta-widget';

/**
 * Date that Site Kit will automatically switch to using GA4: September 25, 2023.
 *
 * With Site Kit's maximum date range being three months, once this period has passed since the UA cutoff date,
 * it will no longer be able to show any historical data for UA and will automatically switch to GA4.
 * For operational reasons, we automatically make the switch a few days before the three month cutoff point.
 *
 * @since 1.107.0
 * @see UA_CUTOFF_DATE.
 */
export const GA4_AUTO_SWITCH_DATE = '2023-09-25';
export const MODULE_SLUG_ANALYTICS_4 = 'analytics-4';
