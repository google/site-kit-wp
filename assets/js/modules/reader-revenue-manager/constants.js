/**
 * Reader Revenue Manager constants.
 *
 * Site Kit by Google, Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const SNIPPET_MODES = {
	post_types: __( 'Specific content types', 'google-site-kit' ),
	per_post: __( 'Specified pages', 'google-site-kit' ),
	sitewide: __( 'Site wide', 'google-site-kit' ),
};

export const RRM_SETUP_NOTIFICATION_ID = 'rrm-setup-notification';
export const RRM_SETUP_SUCCESS_NOTIFICATION_ID =
	'setup-success-notification-rrm';
export const RRM_PRODUCT_ID_CONTRIBUTIONS_NOTIFICATION_ID =
	'rrm-product-id-contributions-notification';
export const RRM_PRODUCT_ID_SUBSCRIPTIONS_NOTIFICATION_ID =
	'rrm-product-id-subscriptions-notification';
