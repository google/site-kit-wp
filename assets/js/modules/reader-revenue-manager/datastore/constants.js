/**
 * `modules/reader-revenue-manager` data store constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const ERROR_CODE_NON_HTTPS_SITE = 'non_https_site';

export const MODULES_READER_REVENUE_MANAGER = 'modules/reader-revenue-manager';

export const READER_REVENUE_MANAGER_MODULE_SLUG = 'reader-revenue-manager';

export const PUBLICATION_ONBOARDING_STATES = {
	ONBOARDING_COMPLETE: 'ONBOARDING_COMPLETE',
	ONBOARDING_ACTION_REQUIRED: 'ONBOARDING_ACTION_REQUIRED',
	PENDING_VERIFICATION: 'PENDING_VERIFICATION',
	UNSPECIFIED: 'ONBOARDING_STATE_UNSPECIFIED',
};

export const UI_KEY_READER_REVENUE_MANAGER_SHOW_PUBLICATION_APPROVED_NOTIFICATION =
	'READER_REVENUE_MANAGER_SHOW_PUBLICATION_APPROVED_NOTIFICATION';

export const LEGACY_RRM_SETUP_BANNER_DISMISSED_KEY =
	'rrm_module_setup_banner_dismissed_key';

export const READER_REVENUE_MANAGER_SETUP_FORM =
	'readerRevenueManagerSetupForm';

export const READER_REVENUE_MANAGER_NOTICES_FORM =
	'readerRevenueManagerNoticesForm';

export const SHOW_PUBLICATION_CREATE = 'showPublicationCreate';

export const RESET_PUBLICATIONS = 'resetPublications';

export const SYNC_PUBLICATION = 'syncPublication';
