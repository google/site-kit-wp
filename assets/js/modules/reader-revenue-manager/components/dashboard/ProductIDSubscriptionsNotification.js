/**
 * ProductIDSubscriptionsNotification component.
 *
 * Site Kit by Google, Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { useSelect } from 'googlesitekit-data';
import { CORE_SITE } from '@/js/googlesitekit/datastore/site/constants';
import NoticeNotification from '@/js/googlesitekit/notifications/components/layout/NoticeNotification';
import Notice from '@/js/components/Notice';
import { MODULE_SLUG_READER_REVENUE_MANAGER } from '@/js/modules/reader-revenue-manager/constants';

export default function ProductIDSubscriptionsNotification( {
	id,
	Notification,
} ) {
	const rrmSettingsURL = useSelect( ( select ) =>
		select( CORE_SITE ).getModuleSettingsEditURL(
			MODULE_SLUG_READER_REVENUE_MANAGER
		)
	);

	return (
		<Notification>
			<NoticeNotification
				notificationID={ id }
				type={ Notice.TYPES.WARNING }
				description={ __(
					'To complete your Reader Revenue Manager paywall setup, add your product IDs in settings',
					'google-site-kit'
				) }
				ctaButton={ {
					label: __( 'Edit settings', 'google-site-kit' ),
					href: rrmSettingsURL,
				} }
				dismissButton
			/>
		</Notification>
	);
}

ProductIDSubscriptionsNotification.propTypes = {
	id: PropTypes.string.isRequired,
	Notification: PropTypes.elementType.isRequired,
};
