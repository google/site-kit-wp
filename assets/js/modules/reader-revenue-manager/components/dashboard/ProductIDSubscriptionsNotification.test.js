/**
 * ProductIDSubscriptionsNotification component tests.
 *
 * Site Kit by Google, Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */

/**
 * Internal dependencies
 */
import {
	createTestRegistry,
	provideModules,
	provideSiteInfo,
	render,
} from '../../../../../../tests/js/test-utils';
import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
import { VIEW_CONTEXT_MAIN_DASHBOARD } from '../../../../googlesitekit/constants';
import { RRM_PRODUCT_ID_SUBSCRIPTIONS_NOTIFICATION_ID } from '../../constants';
import {
	MODULES_READER_REVENUE_MANAGER,
	PUBLICATION_ONBOARDING_STATES,
} from '../../datastore/constants';
import { NOTIFICATIONS } from '../..';
import { withNotificationComponentProps } from '../../../../googlesitekit/notifications/util/component-props';
import ProductIDSubscriptionsNotification from './ProductIDSubscriptionsNotification';

describe( 'ProductIDSubscriptionsNotification', () => {
	let registry;

	const notification =
		NOTIFICATIONS[ RRM_PRODUCT_ID_SUBSCRIPTIONS_NOTIFICATION_ID ];

	const NotificationWithComponentProps = withNotificationComponentProps(
		RRM_PRODUCT_ID_SUBSCRIPTIONS_NOTIFICATION_ID
	)( ProductIDSubscriptionsNotification );

	beforeEach( () => {
		registry = createTestRegistry();

		provideSiteInfo( registry );
		provideModules( registry, [
			{
				slug: 'reader-revenue-manager',
				active: true,
				connected: true,
			},
		] );

		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
	} );

	afterEach( () => {
		jest.clearAllMocks();
	} );

	it( 'should render correctly', () => {
		registry
			.dispatch( MODULES_READER_REVENUE_MANAGER )
			.receiveGetSettings( {
				paymentOption: 'subscriptions',
				productIDs: [],
				productID: 'openaccess',
				publicationOnboardingState:
					PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE,
			} );

		const { container, getByText } = render(
			<NotificationWithComponentProps />,
			{
				registry,
				viewContext: VIEW_CONTEXT_MAIN_DASHBOARD,
			}
		);

		expect(
			getByText(
				'To complete your Reader Revenue Manager paywall setup, add your product IDs in settings'
			)
		).toBeInTheDocument();

		expect( container ).toMatchSnapshot();
	} );

	describe( 'checkRequirements', () => {
		it( 'is active when all required conditions are met', async () => {
			registry
				.dispatch( MODULES_READER_REVENUE_MANAGER )
				.receiveGetSettings( {
					paymentOption: 'subscriptions',
					productIDs: [ '123', '456' ],
					productID: 'openaccess',
					publicationOnboardingState:
						PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE,
				} );

			const isActive = await notification.checkRequirements(
				registry,
				VIEW_CONTEXT_MAIN_DASHBOARD
			);

			expect( isActive ).toBe( true );
		} );
	} );

	it.each( [
		[
			'publicationOnboardingState is not ONBOARDING_COMPLETE',
			{
				paymentOption: 'subscriptions',
				productIDs: [ '123', '456' ],
				productID: 'openaccess',
				publicationOnboardingState:
					PUBLICATION_ONBOARDING_STATES.ONBOARDING_ACTION_REQUIRED,
			},
		],
		[
			'productIDs is empty',
			{
				paymentOption: 'subscriptions',
				productIDs: [],
				productID: 'openaccess',
				publicationOnboardingState:
					PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE,
			},
		],
		[
			'productID is not `openaccess`',
			{
				paymentOption: 'subscriptions',
				productIDs: [ '123', '456' ],
				productID: '123',
				publicationOnboardingState:
					PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE,
			},
		],
		[
			'paymentOption is not `subscriptions`',
			{
				paymentOption: 'contributions',
				productIDs: [ '123', '456' ],
				productID: 'openaccess',
				publicationOnboardingState:
					PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE,
			},
		],
	] )( 'should not be active when %s', async ( _, settings ) => {
		registry
			.dispatch( MODULES_READER_REVENUE_MANAGER )
			.receiveGetSettings( settings );

		const isActive = await notification.checkRequirements(
			registry,
			VIEW_CONTEXT_MAIN_DASHBOARD
		);

		expect( isActive ).toBe( false );
	} );
} );
