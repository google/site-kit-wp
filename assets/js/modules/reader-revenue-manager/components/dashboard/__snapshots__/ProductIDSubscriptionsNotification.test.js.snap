// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductIDSubscriptionsNotification should render correctly 1`] = `
<div>
  <section
    id="rrm-product-id-subscriptions-notification"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--align-middle mdc-layout-grid__cell--span-12"
        >
          <div
            class="googlesitekit-notice googlesitekit-notice--warning"
          >
            <div
              class="googlesitekit-notice__icon"
            >
              <svg />
            </div>
            <div
              class="googlesitekit-notice__content"
            >
              <p
                class="googlesitekit-notice__description"
              >
                To complete your Reader Revenue Manager paywall setup, add your product IDs in settings
              </p>
            </div>
            <div
              class="googlesitekit-notice__action"
            >
              <button
                class="mdc-button mdc-button--tertiary"
                target="_self"
              >
                <span
                  class="mdc-button__label"
                >
                  Got it
                </span>
              </button>
              <a
                class="mdc-button googlesitekit-notice__cta googlesitekit-button-icon--spinner googlesitekit-button-icon--spinner__after mdc-button--raised"
                href="http://example.com/wp-admin/admin.php?page=googlesitekit-settings#connected-services/reader-revenue-manager/edit"
                role="button"
                target="_self"
              >
                <span
                  class="mdc-button__label"
                >
                  Edit settings
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
